File: 03028.py
cup = [1, 0, 0]
shuffle = input()

for i in range(len(shuffle)):
    if shuffle[i] == 'A':
        cup[0], cup[1] = cup[1], cup[0]
    elif shuffle[i] == 'B':
        cup[1], cup[2] = cup[2], cup[1]
    else:
        cup[0], cup[2] = cup[2], cup[0]
        
for i in range(3):
    if cup[i]:
        print(i + 1)
        break

File: 03028.cpp
#include <iostream>
using namespace std;

void swap(int& a, int& b) {
	int tmp = a;
	a = b;
	b = tmp;
}

int main() {
	int cup[3] = { 1,0,0 };
	string shuffle; cin >> shuffle;

	for (int i = 0; i < shuffle.length(); i++) {
		switch (shuffle[i]) {
		case 'A':
			swap(cup[0], cup[1]);
			break;
		case 'B':
			swap(cup[1], cup[2]);
			break;
		case 'C':
			swap(cup[0], cup[2]);
			break;
		}
	}

	for(int i=0;i<3;i++)
		if (cup[i]) {
			cout << i + 1 << endl;
			break;
		}
}
